cfpb/cfgov-refresh
# Filterable Lists: Show a customizable message when there are no posts

Previously, when a Filterable List page had no pages to filter (e.g. the Events page right now), the page would show the filter controls, which didn't make sense, and which had a broken Topic multi-select.

Now, a Wagtail user can specify a two-part message (header and body) in the "no posts message" and "no posts explanation" fields. That message will show up instead of the filter controls when there are no pages to show.

When there are pages to filter, nothing changes.


## Additions

- Added two fields to the Wagtail editing page for Filterable List pages. The "No posts message" and "No posts explanation" fields determine what message will be displayed when there are no pages to filter.

## Changes

- In the Filterable List Controls template, only show the controls if there are any posts to filter. If not, show the message specified in the "No posts message" field in Wagtail.

## Testing

1. Running this branch, visit http://localhost:8000/about-us/events/.
2. See that the filter controls aren't shown.
3. Log in to the Wagtail editor, navigate to the Events page. http://localhost:8000/admin/pages/1090/edit/
4. Enter text in the "No posts message" and "No posts explanation" fields. Save draft and publish.
    1. You might run into an issue where other fields on the page cause validation errors. I got around these by deleting the "Full Width Text" block at the bottom of the page and entering anything in the "Post date description" field.
5. Visit the events page again: http://localhost:8000/about-us/events/. See that your message appears on the page.

## Screenshots

Wagtail:
![screen shot 2019-01-14 at 4 17 19 pm](https://user-images.githubusercontent.com/4295388/51141814-36d94080-1818-11e9-92d2-8de707bdef2b.png)


Event page:
![screen shot 2019-01-14 at 2 12 43 pm](https://user-images.githubusercontent.com/4295388/51134977-f7562880-1806-11e9-905f-8713f98f432e.png)

## Todos

- I'll need to work with someone who has access to Wagtail to make fill in the content in Prod.

## Checklist


## Testing checklist

### Browsers


### Accessibility


### Other
